src/modules/SignUp.js

```diff
@@ -0,0 +1,150 @@
+import { Formik } from "formik";
+import { useHistory } from "react-router-dom";
+import FormInput from '../FormInput';
+import FormButton from '../FormButton';
+
+
+function SignUp() {
+    const initialValues = {
+        name: '',
+        email: '',
+        password: '',
+    };
+
+    let validate = (values) => {
+        let errors = {};
+        const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+        if (!values.userName) {
+            errors.userName = 'Name is required';
+        } else if (values.userName.length > 255) {
+            errors.userName = 'Name too long';
+        }
+        if (!values.email) {
+            errors.email = 'Email is required';
+        } else if (!regex.test(values.email)) {
+            errors.email = 'Invalid Email';
+        } else if (values.email.length > 255) {
+            errors.email = 'Email too long';
+        }
+        if (!values.password) {
+            errors.password = 'Password is required';
+        } else if (values.password.length < 8) {
+            errors.password = 'Password too short';
+        } else if (values.password.length > 255) {
+            errors.password = 'Password too long';
+        }
+        return errors;
+    };
+
+    let history = useHistory();
+
+    function handleLogInClick() {
+        history.push('/SignIn');
+    }
+
+    const submitForm = (values, formikBag) => {
+        const jsonData = {
+            name: values.userName,
+            email: values.email,
+            password: values.password
+        }
+
+        const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(jsonData)
+        };
+
+        fetch('http://localhost:8888/v1/user/sign-up', requestOptions)
+            .then(response => {
+                if (response.status == 201) {
+                    history.push('/SignUpConfirmation');
+                } else if (response.status == 409) {
+                    formikBag.setFieldError('email', 'The user with such email already signed up.');
+                }
+            })
+    };
+
+    return (
+        <Formik
+            initialValues={initialValues}
+            validate={validate}
+            onSubmit={submitForm}
+        >
+            {(formik) => {
+                const {
+                    values,
+                    handleChange,
+                    handleSubmit,
+                    errors,
+                    touched,
+                    handleBlur,
+                    isValid,
+                    dirty
+                } = formik;
+                return (
+                    <>
+                        <div className='header'>
```

Comment:
I believe you need to separate some SinglePanelLayout that encapsulate where and how to show header, and it's content

Reply:
That means I have to create for te header a separate component and then use it in all app views?